catalog/security/catalog-security-uploader-policyplugin/src/main/java/org/codice/ddf/catalog/security/policy/metacard/UploaderPolicyPlugin.java

```diff
@@ -0,0 +1,106 @@
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+
+/**
+ * <p>
+ * Plugin that adds polices to the security map with the metacard's point of contact.
+ * This way, the UploaderPolicyExtension can check the user's email against the metacard's
+ * point of contact and give a security exception if they're equal.
+ * </p>
+ */
+public class UploaderPolicyPlugin implements PolicyPlugin {
+
+    @Override
+    public PolicyResponse processPreCreate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreUpdate(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return getPointOfContactSecurity(input);
+    }
+
+    @Override
+    public PolicyResponse processPreDelete(List<Metacard> metacards,
+            Map<String, Serializable> properties) throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostDelete(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreQuery(Query query, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostQuery(Result input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return getPointOfContactSecurity(input.getMetacard());
+    }
+
+    @Override
+    public PolicyResponse processPreResource(ResourceRequest resourceRequest)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostResource(ResourceResponse resourceResponse, Metacard metacard)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    private PolicyResponse getPointOfContactSecurity(Metacard metacard) {
+        if (metacard.getAttribute(Metacard.POINT_OF_CONTACT) != null) {
+            HashMap<String, Set<String>> securityFinalMap = new HashMap<>();
+            HashSet pocSet = new HashSet(Arrays.asList(metacard
+                    .getAttribute(Metacard.POINT_OF_CONTACT)
+                    .getValue()));
+            securityFinalMap.put(Metacard.POINT_OF_CONTACT + "-all", pocSet);
+            securityFinalMap.put(Metacard.POINT_OF_CONTACT + "-one", pocSet);
+            securityFinalMap.put(Metacard.POINT_OF_CONTACT + "-xacml", pocSet);
```

Comment:
Because of the way `AuthzRealm` implies the permissions. It break permissions out into 3 categories: MatchOne, MatchAll, and PreXACML. In order to imply all permissions from all 3 categories, I needed a permission to check for in each. Unless we change `AuthzRealm` to allow for a Policy Extension that can imply all permissions.